### ISSUE TITLE
Running Flint

### ISSUE THREAD

**Issue by beatrizmoreira** [flintlang/flint#483]
Hello,
I am trying to install and run flint on my mac, but I cannot do it.
I tried through Docker (in your language-guide there's a . (dot) missing at the end of the command `sudo docker build -t "flint_docker" `) and on the [15/17] there is an undeclared type 'PropertyListEncoder' which then results in an error.
I also tried building from source, and I installed the Solidity Compiler but when I do `which solc` nothing appears, only when I do `solcjs`. So I cannot run your code as there is no `solc` command.
Is there another way to do it? 
Thank you!!

![Screenshot 2020-10-08 at 18 22 56](https://user-images.githubusercontent.com/22676800/95599049-8037dd80-0a48-11eb-832b-d690ffd48c49.png)

**Comment by beatrizmoreira** [comment]
Hello!

I have a Mac and I cannot build it. I have installed LLVM and Rust but when i do `cargo build` this error appears:

![Screenshot 2020-10-16 at 09 23 24](https://user-images.githubusercontent.com/22676800/96234441-5a9f5c80-0f91-11eb-9318-4aafbb4ccf7a.png)

I tried installing it through homebrew and cargo but no luck :( 

Thank you !!

**Comment by SusanEisenbach** [comment]
You may be having trouble because your default LLVM installation is the wrong version. If you homebrewed LLVM, I imagine you would have LLVM 10 (which is correct). But you may have another version acting as the default. If you can find the path to your llvm-10 installation, you can compile with ```LLVM_SYS_100_PREFIX=/usr/lib/llvm-10 cargo build```, as seen in the travis file. 

We’re just curious. What are you hoping to do with Flint?

**Comment by beatrizmoreira** [comment]
Ok, I did that and I was able to install flint successfully but when I try to compile your example `Counter` this appears:

<img width="534" alt="Screenshot 2020-10-23 at 08 48 46" src="https://user-images.githubusercontent.com/22676800/96971121-92b41b80-150c-11eb-92a3-4be439b65f8a.png">

And then I remove the parenthesis and I get this error:

<img width="705" alt="Screenshot 2020-10-23 at 08 49 45" src="https://user-images.githubusercontent.com/22676800/96971204-b5463480-150c-11eb-8ee6-a9dbd0d30ed5.png">

Regarding your question, I am trying to formalize Flint's semantic and use Racket to see a step-by-step execution.

**Comment by SusanEisenbach** [comment]
It would be nice to have a formal semantics. We don't actually have a published paper (there is one on Arxiv and a short one right at the begining of Flint) on Flint and if you produce good semantics we would be happy to try to put a Flint paper together for publication. 

George said: You will need to run cargo with the ```LLVM_SYS_100_PREFIX=<path_to_her_llvm> argument``` every time. Otherwise the LLVM API will get confused about which version to use. The brackets should not be used. 

To compile a contract, the syntax (I think) is ```cargo run <libra | ethereum> <path to contract>```. I believe there is a command line help menu, which Matthew wrote. This can be accessed by cargo run -- -h​. Again, she will need to use the LLVM_SYS... each time.

**Comment by beatrizmoreira** [comment]
The goal of our work is to define a small smart language with behavioural types. Flint is a good starting point, since it has typestates; [Featherweight Solidity](http://tesi.cab.unipd.it/61297/) (FS) is another source of inspiration since it is a core language with a formal semantics and a type system.

To better understand a formal language and to debug its definition, we use [Racket](https://racket-lang.org), as it allows to take a formalisation of a programming language and to have a step-by-step visualisation of the formal semantics, by using Redex (one of its many tools). With it, we are able to generate reduction graphs automatically, and see the environments change with the progression of the evaluation of the expressions and check if the formalisations are indeed correct.

Right now we wrote in Racket the formal semantics of FS. It was not super easy, but we managed to achieve the goal. With the lessons learned we intend to formalize Flint, so we can better understand your language. The aim would be then to get the “best” from Flint and FS and add a typestate/behavioural type system, inspired by yours and by approaches like [Mungo](http://www.dcs.gla.ac.uk/research/mungo/).


Also, can you give me an example on where am I supposed to write `LLVM_SYS_100_PREFIX=<path_to_her_llvm>` in the command line to run cargo?

Thank you :)

**Comment by Aurel300** [comment]
`LLVM_SYS_100_PREFIX=<path_to_llvm>` is an environment variable, so you can prepend it to your `cargo` call like so:

```bash
LLVM_SYS_100_PREFIX=<path_to_llvm> cargo run ...
```

You can also make it permanent for the current command-line session by running:

```bash
export LLVM_SYS_100_PREFIX=<path_to_llvm>
```

**Comment by beatrizmoreira** [comment]
Hi,
Thank you! I executed the command `export` and tried to run `cargo run (libra|ethereum) main.flint` like you have in your [Language Guide](https://github.com/flintlang/flint-2/blob/master/docs/guide.md) but this error appears:
 
<img width="492" alt="Screenshot 2020-11-12 at 14 57 23" src="https://user-images.githubusercontent.com/22676800/98955908-5fefb880-24f7-11eb-8efe-b8de90dabec3.png">

So I removed the parenthesis and run `cargo run libra|ethereum main.flint` but this happened :

<img width="490" alt="Screenshot 2020-11-12 at 15 00 25" src="https://user-images.githubusercontent.com/22676800/98956331-cd034e00-24f7-11eb-9b07-07fadfe1e1a5.png">

My version of llvm is 10.0.1_2 .
Thank you for your time :)

**Comment by Aurel300** [comment]
> A Flint source file named `main.flint` containing a contract `Counter` can be compiled to a blockchain-specific IR file using:
> 
> ```
> $ cargo run (libra|ethereum) main.flint
> ```

This part (and any use of the `(libra|ethereum)` syntax) means:

- run `cargo run libra main.flint` to compile for Libra
- run `cargo run ethereum main.flint` to compile for Ethereum

**Comment by beatrizmoreira** [comment]
This happens with either the commands :c


<img width="575" alt="Screenshot 2020-11-13 at 15 40 05" src="https://user-images.githubusercontent.com/22676800/99090435-8deffe00-25c6-11eb-9eb3-e9178e4d1f73.png">

**Comment by Aurel300** [comment]
That still seems to be complaining about no LLVM found. Have you run the `export ...` command I posted above before running `cargo run`? It only applies to the current terminal window and only until it is closed. Out of curiosity, what is your LLVM path?

**Comment by beatrizmoreira** [comment]
Yes, I ran the `export LLVM_SYS_100_PREFIX=usr/local/Cellar/llvm/10.0.1_2` command before `cargo run`.

**Comment by Aurel300** [comment]
Maybe that is the problem, try with a slash in front of `usr`:

```
export LLVM_SYS_100_PREFIX=/usr/local/Cellar/llvm/10.0.1_2 
```

**Comment by beatrizmoreira** [comment]
Thank you, now it compiles but im still not able to run it :c 
<img width="576" alt="Screenshot 2020-11-19 at 13 49 02" src="https://user-images.githubusercontent.com/22676800/99677716-f0874500-2a71-11eb-93eb-af003b2da354.png">
